engine/runner-common/src/main/java/org/enso/runner/common/LanguageServerApi.java

```diff
@@ -0,0 +1,29 @@
+package org.enso.runner.common;
+
+import java.util.ServiceLoader;
+import org.apache.commons.cli.CommandLine;
+import org.slf4j.event.Level;
+
+public abstract class LanguageServerApi {
+  public static final String DAEMONIZE_OPTION = "daemon";
+  public static final String ROOT_ID_OPTION = "root-id";
+  public static final String ROOT_PATH_OPTION = "path";
+  public static final String INTERFACE_OPTION = "interface";
+  public static final String RPC_PORT_OPTION = "rpc-port";
+  public static final String DATA_PORT_OPTION = "data-port";
+  public static final String SECURE_RPC_PORT_OPTION = "secure-rpc-port";
+  public static final String SECURE_DATA_PORT_OPTION = "secure-data-port";
+  public static final String SKIP_GRAALVM_UPDATER = "skip-graalvm-updater";
+
+  public static void launchLanguageServer(CommandLine line, ProfilingConfig config, Level logLevel)
+      throws WrongOption {
+    var it =
+        ServiceLoader.load(LanguageServerApi.class, LanguageServerApi.class.getClassLoader())
+            .iterator();
+    var impl = it.next();
```

Comment:
This is the crux of the change:
- `engine-runner` only knows and calls to `LanguageServerApi`
- the API [looks and implementation up](http://wiki.apidesign.org/wiki/Component_Injection)
- the implementation is registered by `language-server` module
- the `language-server` module knows nothing about `runner`

The `engine-runner` and `language-server` are compile time independent. However they are loosely coupled during runtime. The `engine-runner` needs `language-server` module when `--server` option is used. If `language-server` isn't present, the `impl` here would be `null` and the execution would crash.

Reply:
Could we add a `null` check here and give a more informative error instead of NPE?

Alternatively, I assume we always compile them such that the language server **is** available. Then I'd at least do an `assert`.

What do you think?

Reply:
I was considering better error message as well. That would be done by throwing `WrongOption` saying something like _"--server option isn't supported as essential support is missing"_. However:

If the `language-server` registration is missing or broken, we get test failures like [LanguageServerGatewaySpec](https://github.com/enso-org/enso/actions/runs/9898098961/job/27344121259#step:7:1517). E.g. right now we are safe and we'll find out before integration that something is wrong with the _loose coupling_. Moreover both of these projects still contribute to a single fat `runner.jar` - e.g. it is not possible to decouple them without rebuilding the code.  However as following issue suggests:

- #10010

I would really like to package those two projects into different JAR files. When the time comes, better error message than `NPE` is certainly going to be needed. But right now I think `NPE` is perfectly OK.

Reply:
OK

Reply:
This [@Ignored test](https://github.com/enso-org/enso/pull/10536/files#diff-6ea131d771318eba4b882ef4460dfd7568287668dd471f871fc531cc4636da69R20) has been created in preparation for the future behavior. Let's hope we don't forget to update it when we separate `language-server` project into its own uber JAR.